go/client/logclient.go

```diff
@@ -104,15 +105,29 @@ type GetProofByHashResponse struct {
 	AuditPath [][]byte `json:"audit_path"` // An array of base64-encoded Merkle Tree nodes proving the inclusion of the chosen certificate.
 }
 
+// Logger ...
+type Logger interface {
+	Info(string, ...interface{})
+}
+
+type basicLogger struct{}
+
+func (bl *basicLogger) Info(msg string, args ...interface{}) {
+	log.Printf(msg, args...)
+}
+
 // New constructs a new LogClient instance.
 // |uri| is the base URI of the CT log instance to interact with, e.g.
 // http://ct.googleapis.com/pilot
 // |hc| is the underlying client to be used for HTTP requests to the CT log.
-func New(uri string, hc *http.Client) *LogClient {
+func New(uri string, hc *http.Client, logger Logger) *LogClient {
```

Comment:
This is a breaking change for existing client code.
You could retain the existing two-argument New(), adding this as an overload.
Then have the two-arg New() call through to the three-arg one, supplying 'nil' (or &basicLogger{}) for the third arg.